From 6548aa1396d9c12159ec719a746c9d76e6ac2c03 Mon Sep 17 00:00:00 2001
From: Bharath Bharathi A <bharathbharathia@ami.com>
Date: Mon, 19 May 2025 13:41:20 +0000
Subject: [PATCH 1/1] Create object path for power restore policy one-time

The phosphor-discover-system-state service was failing
because the object path for the one-time restore policy
was not created. This commit resolves the issue by ensuring
the object path is correctly created and handled.

Upstream-Status: Pending

Signed-off-by: Bharath Bharathi A <bharathbharathia@ami.com>
Signed-off-by: Mohammed Javith Akthar M <mohammedjavitham@ami.com>
---
 include/defaults.hpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/include/defaults.hpp b/include/defaults.hpp
index 916d15b..a2a8701 100644
--- a/include/defaults.hpp
+++ b/include/defaults.hpp
@@ -81,6 +81,15 @@ inline void loadSettings(sdbusplus::asio::object_server &objectServer,
     setting->addProperty("PowerCap", static_cast<uint32_t>(0));
     setting->addProperty("PowerCapEnable", false);
 
+    setting = &settings.emplace_back(
+        objectServer, "/xyz/openbmc_project/control/host0/power_restore_policy/one_time",
+        "xyz.openbmc_project.Control.Power.RestorePolicy");
+
+    setting->addProperty(
+        "PowerRestorePolicy",
+        "xyz.openbmc_project.Control.Power.RestorePolicy.Policy.None");
+    setting->addProperty("PowerRestoreDelay", static_cast<uint64_t>(0));
+
     setting = &settings.emplace_back(
         objectServer, "/xyz/openbmc_project/control/host0/power_restore_policy",
         "xyz.openbmc_project.Control.Power.RestorePolicy");
-- 
2.43.0

